dfget/util/file_util.go

```diff
@@ -35,33 +37,75 @@ func CreateDirectory(dirPath string) error {
 
 // DeleteFile deletes a file not a directory.
 func DeleteFile(filePath string) error {
-	return nil
+	return os.Remove(filePath)
 }
 
 // DeleteFiles deletes all the given files.
 func DeleteFiles(filePaths ...string) {
+	if len(filePaths) > 0 {
+		for _, f := range filePaths {
+			if err := os.Remove(f); err != nil {
+				fmt.Printf("failed to remove file %s", f)
+			}
+		}
+	}
 
 }
 
 // OpenFile open a file. If the file isn't exist, it will create the file.
 // If the directory isn't exist, it will create the directory.
 func OpenFile(path string, flag int, perm os.FileMode) (*os.File, error) {
-	return nil, nil
+	if PathExist(path) {
+		return os.OpenFile(path, flag, perm)
+	}
+	pathDir := filepath.Dir(path)
+	// when path is only a file name, e.g: a.txt, the pathDir is current path ".", then just create it
+	if pathDir == "." {
+		return os.OpenFile(path, flag, perm)
+	}
+	if err := os.MkdirAll(pathDir, perm); err != nil {
+		return nil, err
+	}
+	return os.OpenFile(path, flag, perm)
 }
 
 // Link creates a hard link pointing to src named linkName.
 func Link(src string, linkName string) error {
-	return nil
+
+	if PathExist(linkName) {
+		if err := DeleteFile(linkName); err != nil {
+			fmt.Printf("failed to delelte old link %s", linkName)
+			return err
+		}
+	}
+
+	return os.Link(src, linkName)
+
 }
 
 // CopyFile copies the file src to dst.
 func CopyFile(src string, dst string) error {
+	data, err := ioutil.ReadFile(src)
+	if err != nil {
+		fmt.Printf("failed to read file %s", src)
+		return err
+	}
+	// Write data to dst
+	err = ioutil.WriteFile(dst, data, 0755)
+	if err != nil {
+		fmt.Printf("failed to read file %s", dst)
+		return err
+	}
 	return nil
 }
 
 // MoveFile moves the file src to dst.
 func MoveFile(src string, dst string) error {
-	return nil
+	if err := DeleteFile(dst); err != nil {
```

Comment:
It will return an error when the `dst` is not exist and `os.Rename` could not be executed.